File: python/problem003.py
# What is the largest prime factor of the number 600851475143?
from math import floor, sqrt


def factorize(num, min_factor=3):
    return [f for f in range(min_factor, floor(sqrt(num))) if num % f == 0]


factors = factorize(600851475143)
prime_factors = [f for f in factors if not factorize(f)]

print(max(prime_factors))

File: c/problem003.c
// What is the largest prime factor of the number 600851475143?
#include <stdio.h>

int main() {
  long long n = 600851475143;

  // n is odd so we only need to check for odd factors
  for (long long i = 3; i * i <= n; i += 2) {
    while (n % i == 0) {
      n /= i;
    }
  }

  printf("%lld\n", n);

  return 0;
}
